src/rust/ide/src/model/synchronized/execution_context.rs

```diff
@@ -0,0 +1,230 @@
+//! A ExecutionContext model which is synchronized with LanguageServer state.
+
+use crate::prelude::*;
+
+use crate::double_representation::definition::DefinitionName;
+use crate::model::execution_context::StackItem;
+
+use enso_protocol::language_server;
+use json_rpc::error::RpcError;
+
+
+
+// ==============
+// === Errors ===
+// ==============
+
+/// Error returned when module path is invalid, i.e. cannot obtain module name from it.
+#[derive(Clone,Copy,Debug,Fail)]
+#[fail(display="Invalid module path.")]
+pub struct InvalidModulePath {}
```

Comment:
TBH need for such error suggets that we might actually do want to have a newtype over Path for Module paths. (that we can represent that the path is a valid module path on the type level — handling such situation shouldn't be a code specific to execution context)